src/construction.cpp

```diff
@@ -615,13 +632,21 @@ bool player_can_build( player &p, const inventory &pinv, const std::string &desc
     return false;
 }
 
+bool character_has_skill_for( const Character &c, const construction &con )
```

Comment:
Yes, eventually they should use the same structure. I'd see all of them in `requirements`, so that you could have different skills required for different tools (say, fabrication 2 and hacksaw or fabrication 4 and just a knife).